Mathlib/Algebra/Star/Order.lean

```diff
@@ -257,3 +260,11 @@ lemma star_lt_one_iff {x : R} : star x < 1 ↔ x < 1 := by
   simpa using star_lt_star_iff (x := x) (y := 1)
 
 end Semiring
+
+instance Nat.instStarOrderedRing : StarOrderedRing ℕ where
+  le_iff a b := by
+    have : AddSubmonoid.closure (range fun x : ℕ ↦ x * x) = ⊤ :=
+      eq_top_mono
+        (AddSubmonoid.closure_mono <| singleton_subset_iff.2 <| mem_range.2 ⟨1, one_mul _⟩)
+        Nat.addSubmonoid_closure_one
+    simp [this, le_iff_exists_add]
```

Comment:
Why not put this in `Data/Nat/Star.lean` to avoid expanding imports?

Reply:
because there's no meaningful difference between `GroupTheory.Submonoid.Basic` and `GroupTheory.Submonoid.Operations` right now (I tried to make sense of where it's split and couldn't find a sensible justification), so I don't think it's worth bothering using one instead of the other until that's figured out.